### PR TITLE
Add basic API docs for the new search/detail endpoints

### PR DESCRIPTION
**Pull request by diox**

### CODE REVIEW

File: docs/topics/api/overview.rst [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+There are some common API responses that you can expect to receive at times.
+
+.. http:get:: /api/v3/...
+
+    :statuscode 401: Authentication is required or failed.
+    :statuscode 403: You are not permitted to perform this action.
+    :statuscode 404: The requested resource could not be found.
+    :statuscode 500: An unknown error occurred.
+    :statuscode 503:
+        The site is in maintenance mode and this operation is not permitted.
```
Comment (muffinresearch) [inline docs/topics/api/overview.rst]: maybe add  "at this current time" since a 503 should be temporary.

File: docs/topics/api/addons.rst [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+Detail
+------
+
+.. _addon-detail:
+
+This endpoint allows you to fetch a specific add-on by id, slug or guid.
+
+**Not implemented yet**.
+
+.. http:get:: /api/v3/addons/addon/(int:id|string:slug|string:guid)/
```
Comment (mstriemer) [inline docs/topics/api/addons.rst]: Are there any slugs and guids that match? I'm guessing the `{}@` requirement in the guid makes it an invalid slug but just wondering.

File: docs/topics/api/overview.rst [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+
+~~~~~~~~~~
```
Comment (muffinresearch) [inline docs/topics/api/overview.rst]: This heading style is a bit ugly imho (the bit on top of the heading is optional) - but I guess it's already in-use so to be consistent it should stay.

File: docs/topics/api/overview.rst [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+.. code-block:: json
+
+    "name": {
+        "en-US": "Games",
+        "fr": "Jeux",
+        "kn": "ಆಟಗಳು"
+    }
+
+However, for performance sake, if you pass the `lang` parameter to
```
Comment (muffinresearch) [inline docs/topics/api/overview.rst]: "for the sake of performance" maybe? Or just "for performance" even?

File: docs/topics/api/overview.rst [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+a `GET` request, then only the most relevant translation (the specified
+language or the fallback, depending on whether a translation is available in
+the requested language) will be returned as a string.
+
+.. code-block:: json
+
+    "name": "Games"
+
+This behaviour also applies to `POST`, `PATCH` and `PUT` requests: you can
+either submit a object containing several translations, or just a string. If
```
Comment (muffinresearch) [inline docs/topics/api/overview.rst]: s/a object/an object/

File: docs/topics/api/addons.rst (hunk 2 above)
Comment (diox) [inline docs/topics/api/addons.rst]: As you guessed, `{}@` are not permitted in slugs. You could technically have a slug with only numeric characters though, clashing with the id, but that's unlikely, I don't think we need to care at the moment. (Note that this endpoint is not implemented yet, I only included it in the docs to have the basic fields output shown somewhere)

**Comment by muffinresearch** [comment]
lgtm r+wc

**Pull request merged**